boot-mongodb-elasticsearch/src/main/java/com/example/mongoes/config/ChangeStreamStartupListener.java

```diff
@@ -1,31 +1,150 @@
+                                        restaurantChangeStreamEvent.getRaw();
+                                if (Objects.requireNonNull(eventRaw).getOperationType()
+                                        == OperationType.DELETE) {
+                                    if (eventRaw.getDocumentKey() != null) {
+                                        var objectId =
+                                                eventRaw.getDocumentKey()
+                                                        .get("_id")
+                                                        .asObjectId()
+                                                        .getValue()
+                                                        .toString();
+                                        this.restaurantESRepository
+                                                .deleteById(objectId)
+                                                .doOnError(
+                                                        error ->
+                                                                log.error(
+                                                                        "Error deleting restaurant by ID: {}",
+                                                                        error.getMessage()))
+                                                .subscribe();
+                                    } else {
+                                        log.warn(
+                                                "Document key is null for DELETE operation. Cannot delete restaurant from repository.");
+                                    }
+                                }
+                            }
+
+                            this.changeStreamResumeRepository
+                                    .update(restaurantChangeStreamEvent.getBsonTimestamp())
+                                    .log()
+                                    .subscribe();
+                        });
+    }
+
+    private ChangeStreamOptions getChangeStreamOption() {
+        List<ChangeStreamResume> resumeTokenList = getResumeToken();
+        ChangeStreamOptions.ChangeStreamOptionsBuilder changeStreamOptionsBuilder;
+        if (resumeTokenList.isEmpty()) {
+            // Scenario where MongoDb is started freshly hence resumeToken is empty
+            changeStreamOptionsBuilder =
+                    ChangeStreamOptions.builder()
+                            .resumeAt(new BsonTimestamp(Instant.now().getEpochSecond()));
+        } else {
+            changeStreamOptionsBuilder =
+                    ChangeStreamOptions.builder()
+                            .resumeAt(resumeTokenList.getFirst().getResumeTimestamp());
+        }
+        changeStreamOptionsBuilder.returnFullDocumentOnUpdate();
+        return changeStreamOptionsBuilder.build();
+    }
+
+    private List<ChangeStreamResume> getResumeToken() {
+        return this.changeStreamResumeRepository.findAll().toStream().toList();
```

Comment:
@coderabbitai, can you please help me with using this.changeStreamResumeRepository.findFirstByOrderByResumeTimestampDesc() here and make necessary changes to the caller method.

Reply:
@coderabbitai, please avoid block as ours is a full reactive application

Reply:
@coderabbitai, as method signature is changed to `private Mono<ChangeStreamOptions> getChangeStreamOption()` from `private ChangeStreamOptions getChangeStreamOption()` please update caller to this method as well